ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -789,6 +789,10 @@ export const RuntimePanel = fp.flow(
   withCurrentWorkspace(),
   withUserProfile()
 )(({ cdrVersionTiersResponse, workspace, profileState, onClose = () => {}}) => {
+  if (!runtimeStore.get().runtimeLoaded) {
```

Comment:
I believe this approach violates the first rule of hooks: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Of course, if you instead wrapped in another component or HOC.. it would be legal

Reply:
I was actually surprised that React didn't complain about it. I figured it would throw an exception or at least print a warning but it just let it run and it seemed to work.

even so, probably not a good idea to break a library rule so I refactored it into a wrapper that loads a spinner.